conda_smithy/configure_feedstock.py

```diff
@@ -1417,6 +1417,7 @@ def main(
     )
 
     copy_feedstock_content(forge_dir)
+    os.chmod(os.path.join(forge_dir, "build-locally.py"), 0o755)
```

Comment:
This doesn't add the file with modified permissions to git

Reply:
isn't that managed by the call to `commit_changes()` below?

Reply:
Nope. `copy_feedstock_content` copies and git adds the files.

Reply:
That's right. We went through a fair bit of pain getting this right. @nicoddemus, was really helpful in guiding the process (particularly from a Windows perspective). The challenge is we can't rely on permissions at the file system level to propagate to `git`. We just have to ensure `git` is aware of the right permissions.

Would encourage you to look at `feedstock_io.py`, which has a lot of this logic. If you want to have execute permissions, you can use the `set_exe_file` function.

Reply:
@scopatz, this is still an issue, right?

Reply:
Why not change the permission of `build-locally.py` to 755 in the repo and then it will be copied with the same permissions ?

Reply:
Done. Not sure if both methods are needed, but they are implemented